lib/vyasa_web/components/contexts/read.ex

```diff
@@ -67,8 +67,92 @@ defmodule VyasaWeb.Context.Read do
     {:ok, socket}
   end
 
+  #received changes to binding
+
+  def update(
+    %{id: "read", binding: bind = %{verse_id: verse_id}},
+    %{
+      assigns: %{
+        kv_verses: verses,
+        draft_reflector:
+        %Sheaf{
+          marks: [%Mark{state: :draft, verse_id: curr_verse_id} = d_mark | marks]
+        } = draft_reflector
+      }
+    } = socket
+  ) when is_binary(curr_verse_id) and verse_id != curr_verse_id do
+    # binding here blocks the stream from appending to quote
+    #
+    bound_verses =
+      verses
+      |> then(&put_in(&1[verse_id].binding, bind))
+      |> then(&put_in(&1[curr_verse_id].binding, nil))
+
+    updated_draft_mark = d_mark |> Mark.update_mark(%{binding: bind, verse_id: verse_id})
+
+    {:ok,
+     socket
+     |> mutate_verses(curr_verse_id, bound_verses)
+     |> mutate_verses(verse_id, bound_verses)
+     |> assign(draft_reflector: %Sheaf{draft_reflector | marks: [updated_draft_mark | marks]})}
+  end
+
+  # already in mark in drafting state, remember to late bind binding => with a fn()
```

Comment:
ah this was about the resolving on binding and binding context for marks more around navigation mark by mark

evaluating it as the user is navigating made sense

Reply:
more of the nav event being wrapped in a nullary callback then just doing a "binding.jump" 

the use-case here is for things like a nav action button where you jumping through a list of marks and you wanna nav thru them 

though not used here explicitly, this will come in handy soon